Date: Wed, 24 Sep 2025 18:55:18 +0000
Subject: x86: crypto: convert exported crypto symbol into pluggable interface
 for CONFIG_CRYPTO_POLYVAL_CLMUL_NI crypto

Apply Crypto API wrappers to the exported crypto symbol in
CONFIG_CRYPTO_POLYVAL_CLMUL_NI-related crypto to convert them into pluggable
interface.

Signed-off-by: Jay Wang <wanjay@amazon.com>

diff --git a/arch/x86/crypto/Makefile b/arch/x86/crypto/Makefile
--- a/arch/x86/crypto/Makefile
+++ b/arch/x86/crypto/Makefile
@@ -52,7 +52,7 @@ aesni-intel-$(CONFIG_64BIT) += aes-ctr-avx-x86_64.o \
 crypto-objs-$(CONFIG_CRYPTO_GHASH_CLMUL_NI_INTEL) += ghash-clmulni-intel.o
 ghash-clmulni-intel-y := ghash-clmulni-intel_asm.o ghash-clmulni-intel_glue.o
 
-obj-$(CONFIG_CRYPTO_POLYVAL_CLMUL_NI) += polyval-clmulni.o
+crypto-objs-$(CONFIG_CRYPTO_POLYVAL_CLMUL_NI) += polyval-clmulni.o
 polyval-clmulni-y := polyval-clmulni_asm.o polyval-clmulni_glue.o
 
 obj-$(CONFIG_CRYPTO_NHPOLY1305_SSE2) += nhpoly1305-sse2.o
diff --git a/arch/x86/crypto/polyval-clmulni_glue.c b/arch/x86/crypto/polyval-clmulni_glue.c
--- a/arch/x86/crypto/polyval-clmulni_glue.c
+++ b/arch/x86/crypto/polyval-clmulni_glue.c
@@ -171,8 +171,8 @@ static void __exit polyval_clmulni_mod_exit(void)
 	crypto_unregister_shash(&polyval_alg);
 }
 
-module_init(polyval_clmulni_mod_init);
-module_exit(polyval_clmulni_mod_exit);
+crypto_module_init(polyval_clmulni_mod_init);
+crypto_module_exit(polyval_clmulni_mod_exit);
 
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("POLYVAL hash function accelerated by PCLMULQDQ-NI");
